java/flight/flight-sql-jdbc-core/src/main/java/org/apache/arrow/driver/jdbc/client/ArrowFlightSqlClientHandler.java

```diff
@@ -250,6 +273,31 @@ public interface PreparedStatement extends AutoCloseable {
     void close();
   }
 
+  /** A connection is created with catalog set as a session option. */
+  private void setSetCatalogInSessionIfPresent() {
```

Comment:
I think this is good for now, but eventually we may want to reframe this as just "start a session" since I assume there'll be other parameters to set.

Reply:
Agreed. This is rather explicit. I'll be looking to post an issue to cover more generalized session options at some point. This is more a MVP implementation for now.